src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -123,16 +124,7 @@ public override Type GetConcreteType(Type parentType)
 
         public override IEnumerable CreateDerivedEnumerableInstance(JsonPropertyInfo collectionPropertyInfo, IList sourceList, string jsonPath, JsonSerializerOptions options)
         {
-            // Implementing types that don't have default constructors are not supported for deserialization.
```

Comment:
@layomia Slight tweak to what your PR did. This creates instances with...
```csharp
object instance = collectionPropertyInfo.DeclaredTypeClassInfo.CreateObject?.Invoke();
```
...so if instance comes back as null or CreateObject itself was null we throw GetNotSupportedException_SerializationNotSupportedCollection (pre-existing at the bottom of the method).